runtime/src/mem/jemalloc/mem-jemalloc.c

```diff
@@ -47,11 +47,12 @@
 // the performance difference in a playground just to see if it's worth
 // investigating further.
 
-// TODO put these into a struct?
-static void*  heap_base = NULL;
-static size_t heap_size = 0;
-static size_t cur_heap_offset = 0;
-static pthread_mutex_t chunk_alloc_lock;
+static struct shared_heap {
+  void* base;
+  size_t size;
+  size_t cur_offset;
+  pthread_mutex_t alloc_lock;
+} heap; // static, will be "zero" initialized automatically
```

Comment:
I think I'm still confused. I get that the PTHREAD_MUTEX_INITIALIZER would allow me to avoid the `/*default init mutex*/,` but that would actually initialize it, whereas I really just want it zeroed.

If I did add the static init, we'd always have to destroy the mutex even when `heap.base == NULL` which seems to make the conditional dynamic initialization the redundant init

Anyways, we're probably saying the same thing and I'm happy leaving the initializer off. I was just noting why I didn't do an explicit initializer, since I figured you might wonder while reviewing.